core/services/job/common.go

```diff
@@ -1,19 +1,54 @@
 package job
 
 import (
+	"context"
 	"net/url"
 	"time"
 )
 
 //go:generate mockery --name Service --output ./mocks/ --case=underscore
+//go:generate mockery --name ServiceCtx --output ./mocks/ --case=underscore
 
 type Service interface {
 	Start() error
 	Close() error
 }
 
+// ServiceCtx is the same as Service, but Start method receives a context.
+type ServiceCtx interface {
+	Start(context.Context) error
+	Close() error
+}
+
 type Config interface {
 	DatabaseURL() url.URL
 	TriggerFallbackDBPollInterval() time.Duration
 	LogSQL() bool
 }
+
+// ServiceAdapter is a helper introduced for transitioning from Service to ServiceCtx.
+type ServiceAdapter interface {
+	ServiceCtx
+}
+
+type adapter struct {
+	service Service
+}
+
+// NewServiceAdapter creates an adapter instance for the given Service.
+func NewServiceAdapter(service Service) ServiceCtx {
```

Comment:
I wonder if 'adapter' is too generic. Is this more clear?
```suggestion
// NewServiceCtx wraps service to implement ServiceCtx.
func NewServiceCtx(service Service) ServiceCtx {
```
And similarly `s/NewOCR2ProviderAdapter/NewOCR2ProviderCtx`?

Reply:
Fixed

Reply:
I see the new ones, but is this old one still needed separately too?